### PR TITLE
Add role management

### PR DESCRIPTION
**Pull request by RobPando**

### CODE REVIEW

File: app/policies/membership_policy.rb [hunk 1]
```diff
@@ -2,7 +2,7 @@ class MembershipPolicy < ApplicationPolicy
   alias membership record
 
   def update?
-    user.administrates_fraternity?(membership.fraternity_id)
+    user.owns_fraternity?(record.fraternity_id)
```
Comment (pablocrivella) [inline app/policies/membership_policy.rb]: Use `membership` instead of `record`

File: app/policies/fraternity_policy.rb [hunk 2]
```diff
@@ -12,7 +12,7 @@ def create?
   end
 
   def update?
-    user.owns_fraternity?(record)
+    user.administrates_fraternity?(record)
```
Comment (pablocrivella) [inline app/policies/fraternity_policy.rb]: Add an alias of `fraternity` for `record` and use it instead so this becomes more clear.

File: spec/policies/membership_policy_spec.rb [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+        it('grants access') {
+          is_expected.to permit(fraternity.owner.user, record)
+        }
+      end
+    end
+  end
+
+  context 'as admin' do
+    let(:user) { create(:user) }
+    let(:admin) { create(:membership, user: user, role: :admin, fraternity: fraternity) }
```
Comment (pablocrivella) [inline spec/policies/membership_policy_spec.rb]: This `let(:admin)` is a bit confusing, i would expect it to be a `User` and not a `Membership`

File: spec/policies/membership_policy_spec.rb [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+
+    permissions :update? do
+      it('denies access') { is_expected.to_not permit(admin.user, record) }
+    end
+  end
+
+  %i[member].each do |role|
+    context "as #{role}" do
+      let(:user) { create(:user) }
+      let(:member) { build(:membership, user: user, role: role, fraternity: fraternity) }
```
Comment (pablocrivella) [inline spec/policies/membership_policy_spec.rb]: Same here

**Comment by RobPando** [comment]
@pablocrivella updated!

**Pull request merged**